src/params.rs

```diff
@@ -50,7 +39,7 @@ impl Default for Params {
 }
 
 pub fn parse_params<I: IntoIterator<Item = OsString>>(opts: I) -> Result<Params, String> {
```

Comment:
the function starts to be a bit long, could you please split it a bit?

Reply:
I shrank it by 33 lines by breaking out the logic to match context/unified diff params into separate functions.
The main function that parse the parameters is still long-ish, but I think it's better in terms of readability.